ctk/common/src/main/kotlin/org/codice/compliance/utils/report/Report.kt

```diff
@@ -0,0 +1,241 @@
+/*
+Copyright (c) 2018 Codice Foundation
+
+Released under the GNU Lesser General Public License; see
+http://www.gnu.org/licenses/lgpl.html
+*/
+package org.codice.compliance.utils.report
+
+import org.codice.compliance.QUIET_MODE
+import org.codice.compliance.SAMLComplianceException
+import org.codice.compliance.utils.report.Report.Section.GENERAL
+import org.codice.compliance.utils.report.Report.Section.SCHEMA
+import org.fusesource.jansi.Ansi
+import java.io.File
+import java.io.PrintWriter
+
+object Report {
+
+    internal const val REPORT_FILE = "report.txt"
+    private const val NOT_TESTED_LIST =
+            "https://github.com/codice/saml-conformance/blob/master/ctk/idp/NotTested.md"
+
+    private val emptyMap: MutableMap<Section, MutableSet<SAMLComplianceException>?> =
+            Section.values().map {
+                it to null
```

Comment:
I had it as null to be able to support skipping a section. Otherwise, if a section is not run, it will mark it as passed which I thought would be misleading.